cmd/healthbot/counterprobe/counterprobe.go

```diff
@@ -144,57 +132,39 @@ func (cp *CounterProbe) healthCheck(ctx context.Context) (int64, error) {
-	updateCounterReq := rpcservice.RelayWriteQueryRequest{
-		Statement: fmt.Sprintf("update %s set counter=counter+1", cp.tableName),
+	hash, err := cp.client.Write(
+		ctx,
+		fmt.Sprintf("update %s set counter=counter+1", cp.tableName),
+		client.WriteRelay(true),
+	)
+	if err != nil {
+		return fmt.Errorf("calling client write: %s", err)
 	}
-	var updateCounterRes rpcservice.RelayWriteQueryResponse
-	if err := cp.rpcClient.CallContext(ctx, &updateCounterRes, "tableland_relayWriteQuery", updateCounterReq); err != nil {
-		return fmt.Errorf("calling tableland_runReadQuery: %s", err)
+	start := time.Now()
+	receipt, ok, err := cp.client.Receipt(ctx, hash, client.WaitFor(cp.receiptTimeout))
+	if err != nil {
+		return fmt.Errorf("getting receipt: %s", err)
 	}
-
-	getReceiptRequest := rpcservice.GetReceiptRequest{
-		TxnHash: updateCounterRes.Transaction.Hash,
+	if !ok {
+		return errors.New("receipt not found before timeout")
 	}
-
-	start := time.Now()
-	deadline := time.Now().Add(cp.receiptTimeout)
-	for time.Now().Before(deadline) {
-		var getReceiptResponse rpcservice.GetReceiptResponse
-		if err := cp.rpcClient.CallContext(ctx, &getReceiptResponse, "tableland_getReceipt", getReceiptRequest); err != nil {
-			return fmt.Errorf("calling tableland_getReceipt: %s", err)
-		}
-		if getReceiptResponse.Ok {
-			if getReceiptResponse.Receipt.Error != "" {
-				return fmt.Errorf("receipt found but has an error %s", getReceiptResponse.Receipt.Error)
-			}
-			cp.log.Info().Int64("duration", time.Since(start).Milliseconds()).Msg("receipt confirmed")
-			return nil
-		}
-		time.Sleep(time.Second * 5)
+	if receipt.Error != "" {
+		return fmt.Errorf("receipt found but has an error %s", receipt.Error)
 	}
-
-	return fmt.Errorf("timed out waiting for receipt %s", getReceiptRequest.TxnHash)
+	cp.log.Info().Int64("duration", time.Since(start).Milliseconds()).Msg("receipt confirmed")
+	return nil
 }
 
 func (cp *CounterProbe) getCurrentCounterValue(ctx context.Context) (int64, error) {
-	output := "table"
-	getCounterReq := rpcservice.RunReadQueryRequest{
-		Statement: fmt.Sprintf("select * from %s", cp.tableName),
-		Output:    &output,
-	}
-
-	type data struct {
-		Rows [][]int64 `json:"rows"`
-	}
-	var getCounterRes struct {
-		Result data `json:"data"`
-	}
-	if err := cp.rpcClient.CallContext(ctx, &getCounterRes, "tableland_runReadQuery", getCounterReq); err != nil {
-		return 0, fmt.Errorf("calling tableland_runSQL: %s", err)
-	}
-	if len(getCounterRes.Result.Rows) != 1 || len(getCounterRes.Result.Rows[0]) != 1 {
-		return 0, fmt.Errorf("unexpected response format")
-	}
-
-	return getCounterRes.Result.Rows[0][0], nil
+	var counter int64
+	if err := cp.client.Read(
+		ctx,
+		fmt.Sprintf("select counter from %s", cp.tableName),
+		&counter,
+		client.ReadExtract(),
+		client.ReadUnwrap(),
+	); err != nil {
+		return 0, fmt.Errorf("calling client read: %s", err)
+	}
+	return counter, nil
```

Comment:
Much cleaner using the client and just unmarshaling directly into our `counter` value.